GetDKAN/dkan
# Improve drop/import command to make filefetcher optional

Simplifys some of the drop/import services and related drush commands. More importantly, introduces a `--keep-local` switch for `drush dkan:datastore:drop` which leaves local file and filefetcher status in tact and only drops the datastore table and import status. This can be useful for working with very large remote files, where one wishes to re-try an import but not have to re-download (re-localize) the file.

## QA Steps

1. Create a new dataset (replace domain and credentials as needed):

```http
POST http://reimport.localtest.me/api/1/metastore/schemas/dataset/items
Authorization: Basic admin:admin

{
  "title": "Reimport test",
  "description": "Testing import commands",
  "modified": "2022-06-01",
  "keyword": ["tag1"],
  "distribution": [
    {
      "title": "Test dirstro",
      "mediaType": "text/csv",
      "downloadURL": "https://people.sc.fsu.edu/~jburkardt/data/csv/hooke.csv"
    }
  ]
}
```

2. Run `drush queue:run datastore_import`
3. Confirm datastore table added
4. Run `dkan:datastore:drop aeca64555b36f41b15b22f6c91c3be9a` (confirm resource identifier is correct)
5. Confirm that datastore table now gone from db, and that the file has been removed from sites/default/files/resources. 
6. Confirm both jobstore results are gone in DB:
```sql
SELECT * FROM jobstore_filefetcher_filefetcher;
SELECT * FROM  jobstore_dkan_datastore_importer;
```
7. Re-import the resource: `drush dkan:datastore:import aeca64555b36f41b15b22f6c91c3be9a`
8. Confirm table has been added to DB.
9. Drop datastore but keep file:  `dkan:datastore:drop aeca64555b36f41b15b22f6c91c3be9a --keep-local`
10. Confirm that datastore table now gone from db, but file is still present in sites/default/files/resources. 
11. Confirm filefetcher results are still present in DB, but importer is gone:
```sql
SELECT * FROM jobstore_filefetcher_filefetcher;
SELECT * FROM  jobstore_dkan_datastore_importer;
```
12. Import again: `drush dkan:datastore:import aeca64555b36f41b15b22f6c91c3be9a`
13. Confirm table has been added to DB.
14. Drop datastore one more time:  `dkan:datastore:drop aeca64555b36f41b15b22f6c91c3be9a`
15. Import with new "deferred" switch: `drush dkan:datastore:import aeca64555b36f41b15b22f6c91c3be9a --deferred`
16. Run `drush queue:run datastore_import`
17. Confirm table added to datastore.
